server/controllers/profileController.js

```diff
@@ -141,4 +144,92 @@ exports.getAllProfiles = async (req, res, next) => {
   } catch (err) {
     next(err);
   }
-};
+});
+
+// @route GET /profile/search
+// @desc Search by city or date ranges for sitters
+// @access Private
+exports.searchSitters = asyncHandler(async (req, res, next) => {
```

Comment:
Optional: It would've been nice to reuse the existing route with query parameters to fetch all the profiles as opposed to creating a new route :)

Existing route: `router.get("/", protect, profileController.getAllProfiles);`

Reply:
I will try it later 😭